crawl4ai/processors/pdf/__init__.py

```
@@ -138,22 +145,45 @@ def _get_pdf_path(self, url: str) -> str:
                     for chunk in response.iter_content(chunk_size=8192):
                         f.write(chunk)
+                        downloaded += len(chunk)
+                        if self.logger and total_size > 0:
+                            progress = (downloaded / total_size) * 100
+                            if progress % 10 < 0.1:  # Log every 10%
+                                self.logger.debug(f"PDF download progress: {progress:.0f}%")
+                
+                if self.logger:
+                    self.logger.info(f"PDF downloaded successfully: {temp_file.name}")
                         
                 return temp_file.name
```

Ruff (0.14.0)

Consider moving this statement to an `else` block

(TRY300)